cli/register/servicewait.go

```diff
@@ -24,115 +105,193 @@ func WaitForService(org string, waitService string, waitTimeout int, pattern str
+	for i, ss := range servSpecArr {
+		if ss.status <= 0 {
+			found := false
+			for _, serviceInstance := range services.Instances["active"] {
+				// 1. Maybe the service is there but just hasnt started yet.
+				if serviceInstance.SpecRef == ss.name && serviceInstance.Org == ss.org {
+					found = true
 
-			// 2. Maybe the service has encountered an error.
-			if serviceInstance.ExecutionStartTime == 0 && serviceInstance.ExecutionFailureCode != 0 {
-				msgPrinter.Printf("Service %v/%v execution failed: %v.", org, waitService, serviceInstance.ExecutionFailureDesc)
-				msgPrinter.Println()
-				serviceFailed = true
-			} else {
-				msgPrinter.Printf("Service %v/%v might need more time to start executing, continuing analysis.", org, waitService)
-				msgPrinter.Println()
+					// 2. Maybe the service has encountered an error.
+					if serviceInstance.ExecutionStartTime == 0 && serviceInstance.ExecutionFailureCode != 0 {
+						failedArr = append(failedArr, ss)
+						failedDescArr = append(failedDescArr, serviceInstance.ExecutionFailureDesc)
+						servSpecArr[i].status = -1
+					} else {
+						delayedArr = append(delayedArr, ss)
+					}
+					break
+				}
 			}
-			break
 
+			// 3. The service might not even be there at all.
+			if !found {
+				notDeployedArr = append(notDeployedArr, ss)
+			}
 		}
 	}
 
-	// 3. The service might not even be there at all.
-	if !found {
-		msgPrinter.Printf("Service %v/%v is not deployed to the node, continuing analysis.", org, waitService)
+	// Print the status of each service
+	if len(failedArr) > 0 {
+		msgPrinter.Printf("The following services execution failed:")
 		msgPrinter.Println()
+		for i, ss := range failedArr {
+			msgPrinter.Printf("\t%v/%v: %v", ss.org, ss.name, failedDescArr[i])
+			msgPrinter.Println()
+		}
 	}
 
-	// 4. Are there any agreements being made? Check for only non-archived agreements. Skip this if we know the service failed
-	// because we know there are agreements.
-	if !serviceFailed {
+	if len(delayedArr) > 0 {
+		msgPrinter.Printf("The following services might need more time to start executing, continuing analysis:")
 		msgPrinter.Println()
-		ags := agreement.GetAgreements(false)
-		if len(ags) != 0 {
-			msgPrinter.Printf("Currently, there are %v active agreements on this node. Use `hzn agreement list' to see the agreements that have been formed so far.", len(ags))
+		for _, ss := range delayedArr {
+			msgPrinter.Printf("\t%v/%v", ss.org, ss.name)
 			msgPrinter.Println()
-		} else {
-			msgPrinter.Printf("Currently, there are no active agreements on this node.")
+		}
+	}
+
+	if len(notDeployedArr) > 0 {
+		msgPrinter.Printf("The following services are not deployed to the node, continuing analysis:")
+		msgPrinter.Println()
+		for _, ss := range notDeployedArr {
+			msgPrinter.Printf("\t%v/%v", ss.org, ss.name)
 			msgPrinter.Println()
 		}
 	}
 
+	// 4. Are there any agreements being made? Check for only non-archived agreements. Skip this if we know the service failed
```

Comment:
I think this section should only run when delayedArr OR notDeployedArr are not empty. I think that's consistent with the behavior of the code before your change.

Reply:
Done.